plugin/core/rpc.py

```diff
@@ -126,42 +111,42 @@ def reset(self) -> None:
         self.__response_id = -1
 
 
-class Client(object):
-    def __init__(self, transport: Transport, settings: Settings) -> None:
-        self.transport = transport  # type: Optional[Transport]
-        self.transport.start(self.receive_payload, self.on_transport_closed)
+def print_to_status_bar(error: Dict[str, Any]) -> None:
+    sublime.status_message(error["message"])
+
+
+def method2attr(method: str) -> str:
+    # window/messageRequest -> m_window_messageRequest
```

Comment:
`m_window_messageRequest` is nice, but it would be nicer if it converted `window/messageRequest -> m_window_message_request` instead  the capital `R` :) 

```
>>> s = "m_window_messageRequest"
>>> ''.join('_' + char.lower() if char.isupper() else char
...               for char in s)
'm_window_message_request'
```

Reply:
I know there's not a request method like this at the moment, but maybe in the future :)

```
>>> s = 'm_window_getHHTPHeader'; ''.join('_' + char.lower() if char.isupper() else char for char in s)
'm_window_get_h_h_t_p_header'
```

Reply:
Can we store a map of request to handler (just a string map so it doesn't affect GC)?
This automatic mapping is not that clear how it works from someone looking at implementation of the plugin.

Reply:
I don't understand what is complicated about it. I'll add more docs in the AbstractPlugin class... For a request/notification foo/bar, replace non-alphabetic characters with _, and prepend m_.

```python
def m_vscode_content(self, params, request_id) -> None:
    """handles vscode/content request"""
    session = self.weaksession()
    if session:
        session.send_response(request_id, ...)

def m_intelephense_analysisStarted(self, params) -> None:
    """handles intelephense/analysisStarted notification"""
    pass

def m_intelephense_analysisStopped(self, params) -> None:
    """handles intelephense/analysisStopped notification"""
    pass

def m_eslint_openDoc(self, params, request_id) -> None:
    """handles eslint/openDoc request"""
    session = self.weaksession()
    if session:
        session.send_response(request_id, ...)

def m_eslint_status(self, params) -> None:
    """handles eslint/status notification"""
    pass
```

Reply:
There is something about it that makes it kinda magic. It would feel better to me to have an imperative API, I think. But I'll resolve it as I don't have any good arguments as to why. :)

Reply:
The magic is in the `__getattr__` method of `Session` :)